tests/tracer/test_sampler.py

```diff
@@ -315,32 +315,41 @@ def test_sampling_rule_init_via_env():
     with pytest.raises(ValueError) as excinfo:
         with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":2.0,"service":"xyz","name":"abc"}]')):
             sampling_rule = DatadogSampler().rules
-    assert "SamplingRule(sample_rate={!r}) must be greater than or equal to 0.0 and less than or equal to 1.0" in str(
+    assert "SamplingRule(sample_rate=2.0) must be greater than or equal to 0.0 and less than or equal to 1.0" in str(
```

Comment:
you could do `str(execinfo.value).endswith("SamplingRule")` (or `.startswith`) if you wanted to make a bolder assertion to where it lives in the exception message.

the reason for this is ensure that nothing comes after it (or before it).